From 4f899efc13a3a8f5259ce260655dfdd6f4830299 Mon Sep 17 00:00:00 2001
From: Alexandre Janniaux <ajanni@videolabs.io>
Date: Fri, 28 Aug 2020 10:08:37 +0200
Subject: [PATCH 1/1] qt: add missing include for QPainterPath

The include is not transitively included anymore with Qt 5.15.

Original contribution of menaquinone from ticket #24882.

Fix #24882, #25059
---
 modules/gui/qt/components/playlist/views.cpp | 1 +
 modules/gui/qt/util/timetooltip.cpp          | 1 -
 modules/gui/qt/util/timetooltip.hpp          | 1 +
 3 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/gui/qt/components/playlist/views.cpp b/modules/gui/qt/components/playlist/views.cpp
index ecc6b9918d..70dd29df72 100644
--- a/modules/gui/qt/components/playlist/views.cpp
+++ b/modules/gui/qt/components/playlist/views.cpp
@@ -27,6 +27,7 @@
 #include "input_manager.hpp"                      /* THEMIM */
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRect>
 #include <QStyleOptionViewItem>
 #include <QFontMetrics>
diff --git a/modules/gui/qt/util/timetooltip.cpp b/modules/gui/qt/util/timetooltip.cpp
index 7a63a9d7d9..8cfdab3206 100644
--- a/modules/gui/qt/util/timetooltip.cpp
+++ b/modules/gui/qt/util/timetooltip.cpp
@@ -23,7 +23,6 @@
 
 #include <QApplication>
 #include <QPainter>
-#include <QPainterPath>
 #include <QBitmap>
 #include <QFontMetrics>
 #include <QDesktopWidget>
diff --git a/modules/gui/qt/util/timetooltip.hpp b/modules/gui/qt/util/timetooltip.hpp
index b6d7c646c9..f213eac459 100644
--- a/modules/gui/qt/util/timetooltip.hpp
+++ b/modules/gui/qt/util/timetooltip.hpp
@@ -25,6 +25,7 @@
 #include "qt.hpp"
 
 #include <QWidget>
+#include <QPainterPath>
 
 class TimeTooltip : public QWidget
 {
-- 
2.20.1

